gamedev-iitk/tower-defense
# Implement a basic form of the tower upgrade system

**Description:** This PR implements a basic prototype for the tower upgrade system. Better datastructures could be used to implement this later on.

**Steps to test:**
1. Start the game and click on a block. Tower upgrades UI should pop up.
1. Click on an option to upgrade your tower to that option.
1. Press escape to close the UI.

**Outstanding:**
1. There is no upgrade hierarchy now, but a way to implement that has been included. Use `setButtonActivation()` and `UpgradeTree`. Issue #4 opened.
